src/tvheadend/HTSPConnection.cpp

```diff
@@ -144,7 +144,13 @@ std::string HTSPConnection::GetWebURL ( const char *fmt, ... ) const
   if (!auth.empty())
     auth += "@";
 
-  std::string url = StringUtils::Format("http://%s%s:%d", auth.c_str(), settings.GetHostname().c_str(), settings.GetPortHTTP());
+  const char *proto = nullptr;
```

Comment:
Could be shortened to `const std::string proto = settings.GetUseHTTPS() ? 'https' : 'http';`